src/shared/redux/slices/publicationsSlice.ts

```diff
@@ -1,57 +1,113 @@
-import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { createSlice, current, PayloadAction } from '@reduxjs/toolkit';
 import { RootState } from '../rootReducer';
 
-type CollaboratorElement = {
-  github_username: string;
+type Collaborator = {
+  id: string;
+  githubUsername: string;
   role: string;
 };
 
-type PublicationListElement = {
-  project_name: string;
-  collaborators: [CollaboratorElement];
-  pm_preference: string;
-  description: string;
+type Publication = {
+  id: string;
   dirPath: string;
+  publicationName: string;
+  description: string;
+  collaborators: Collaborator[];
+  packageManager: string;
+  useSass: string;
+  useTypescript: string;
```

Comment:
Just noticed now. 🤔 Wouldn't it make more sense for these two properties to be `boolean`?